modules/serverless-ossm-setup-without-kourier-mtls.adoc

```diff
@@ -0,0 +1,208 @@
++
+.Example `knative-local-gateway` using HTTPS
+[source,yaml]
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: knative-local-gateway
+  namespace: knative-serving
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+    - port:
+        number: 443
+        name: https
+        protocol: HTTPS
+      hosts:
+        - "*"
+      tls:
+        mode: SIMPLE
+        credentialName: <wildcard_certs>
+----
+. Apply `Gateway` resources:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+. Install Knative Serving by creating the `KnativeServing` custom resource definition (CRD), and enable Istio integration:
++
+[source,yaml]
+----
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+spec:
+  ingress:
+    istio:
+      enabled: true <1>
+  deployments: <2>
+  - name: activator
+    annotations:
+      "sidecar.istio.io/inject": "true"
+      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
+  - name: autoscaler
+    annotations:
+      "sidecar.istio.io/inject": "true"
+      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
+----
+<1> Enables Istio integration.
+<2> Enables sidecar injection for Knative Serving data plane pods.
+. Apply the `KnativeServing` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+. Create a Knative Service that has sidecar injection enabled and uses a pass-through route:
++
+[source,yaml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: <service_name>
+  namespace: default
+  annotations:
+    serving.knative.openshift.io/enablePassthrough: "true" <1>
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "true" <2>
+        sidecar.istio.io/rewriteAppHTTPProbers: "true"
+    spec:
+      containers:
+      - image: <image_url>
+----
+<1> Instructs Knative Serving to generate an OpenShift pass-through enabled route, so that the certificates you have generated are served through the ingress gateway directly.
+<2> Injects {ProductShortName} sidecars into the Knative service pods.
+. Apply the `Service` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+.Verification
+
+* Access your serverless application by a secure connection that is now trusted by the CA:
++
+[source,terminal]
+----
+$ curl --cacert root.crt <service_url>
+----
++
+.Example command
+[source,terminal]
+----
+$ curl --cacert root.crt \
+    --header "Host: hello-default.example.com" \
+    --resolve "hello-default.example.com:443:${ROUTER_IP}" \
+     https://hello-default.example.com
```

Comment:
This doesn't work for me @nak3 

If I call the ksvc URL (the https version), I can see the `*.example.com` certificate presented. If I call it like this, the standard no application OCP page is presented.

Even if I create a `VirtualService` like this, it doesn't work either:

```yaml
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  name: showcase
  namespace: demo
spec:
  gateways:
    - default-gateway
  hosts:
    - example.com
    - '*.example.com'
  http:
    - rewrite:
        authority: showcase.demo.svc
      route:
        - destination:
            host: showcase.demo.svc
            port:
              number: 80
```

Reply:
@cardil Could you please quickly verify if your pod got istio injection like this?
```
$ oc get pod -n serving-tests
NAME                                      READY   STATUS    RESTARTS   AGE
hello-00001-deployment-6fc89765b8-tlnqc   3/3     Running   0          28s
```
If yes, I think we need to look into it more, so can we discuss it on the Slack?

Reply:
Yes. I had istio proxy container. 3 containers total.

Reply:
Do we even need the `--resolve` and --header stuff? In my tests, `curl --cacert wildcard.crt https://hello-default.example.com` worked just fine. Or is that because I used the wildcard and not the root cert?

Reply:
Okay. We discussed this with @nak3.

I think we should add to this doc, a step to deploy a `DomainMapping` for `*.example.com`, so that cluster could support a wildcard cert we are generating in this doc.

An alternative would be to constrain the doc to use cluster base cert name instead of `example.com`, but that has massive downsides in my opinion:

1. It's pointless as that wildcard is already supported by OpenShift out of the box. If user deploys own certs for the whole cluster.
2. No one should publish an OCP cluster to the public domain, and have an OCP console just beside the production websites. That would be if a `*.example.com` would be set as cluster base name.

Reply:
FWIW, usually the user would have DNS setup for `*.example.com` (which is none of our business), which in this case would allow them to simplify this to

```
curl --cacert root.crt https://hello-default.example.com
```

right?

Reply:
Yes. If DNS is properly configured.

Reply:
Updating this to use Markus's suggestion. If there's anything else required here please decide on the steps that need to be added among yourselves and let me know as a suggestion on this PR / open a new issue.